Source/WebKit/UIProcess/WebProcessProxy.h

```diff
@@ -495,6 +498,8 @@ class WebProcessProxy : public AuxiliaryProcessProxy, private ProcessThrottlerCl
 
     Logger& logger();
 
+    bool hasMediaActivity() const { return m_audibleMediaActivity.has_value() || m_mediaStreamingActivity.has_value(); }
```

Comment:
This feels like a hack? As I remember it, we drop those activities on a timer (in case new media starts soon after).
If a process goes into the process cache, we should just drop those media activities, not throw away the whole process.